### ISSUE TITLE
freeform/plugin.js is adding event listeners to non-freeform forms

### ISSUE THREAD

**Issue by sgtpenguin** [solspace/craft3-freeform#1416]
### What happened?

I am including the freeform/plugin.js script manuallly to enable ajax forms on a Craft headless NextJs setup.  This site also has html `form`s that exist apart from freeform and freeform is adding event listeners to these forms which is preventing the execution of my own `onSubmit` handler.  This problem only popped up after ugrading from Craft 3 to Craft 4 a few days ago.

### How can we reproduce this?

1. include plugin.js manually
2. create a random form with onsubmit handler

### Freeform Edition

Lite

### Freeform Version

5.5.1

### Craft Version

4.10.3

### When did this issue start?

After upgrading from older Freeform version, After upgrading from older Craft version

### Previous Freeform Version

3.13.37

**Comment by gustavs-gutmanis** [comment]
Hi @sgtpenguin,

Freeform JS plugin only attaches to forms that have the `data-freeform` attribute on the `<form>` element. 

```ts
const forms = document.querySelectorAll<HTMLFormElement>('form[data-freeform]');
forms.forEach((form) => {
  new Freeform(form);
});
```

```ts
const recursiveFreeformAttachment = (node: HTMLFormElement) => {
  if (node.nodeName === 'FORM' || node.dataset?.freeform !== undefined) {
    new Freeform(node);
  }

  node?.childNodes.forEach(recursiveFreeformAttachment);
};
```

Is it at all possible that you have that attribute set on your non-freeform forms?

**Comment by sgtpenguin** [comment]
Thanks for getting back to me.  That makes sense, but no I don't have that attribute on the form.  Here's a screenshot with the Event Listeners attached:
<img width="1511" alt="Screenshot 2024-07-11 at 8 30 06 AM" src="https://github.com/solspace/craft-freeform/assets/16600959/c295f269-2c8f-4d4e-856e-404a5ccde63f">
If you want to inspect it [here's a page, click "enroll"](https://civilla.org/practica/foundations).  In my case, the conflicting submit event was preventing my submit handler from working.  I have found a workaround by setting the 3rd param true in my submit handler `form.addEventListener("submit", onSubmit, true);` in combination with `event.stopPropagation` and this seems to let me take priority over the plugin.js handler.

**Comment by gustavs-gutmanis** [comment]
Thanks for reporting this, super weird. I will look into it and fix it.

**Comment by gustavs-gutmanis** [comment]
Sorry for the long wait @sgtpenguin, the issue was a simple oversight on my part in the logic which checks if the plugin should be loaded on the form or not.  
The fix will be released with the upcoming release.

**Comment by sgtpenguin** [comment]
Amazing.  Thank you for following up and fixing!

**Comment by kjmartens** [comment]
This should be resolved in Freeform 5.5.7 now. 🙂
